packages/account/imAccount/validator/WebAuthnOwner.ts

```diff
@@ -0,0 +1,88 @@
+import {
+  isoBase64URL,
+  isoUint8Array,
+  parseAuthenticatorData
+} from "@simplewebauthn/server/helpers"
+import * as ethers from "ethers"
+import browser from "webextension-polyfill"
+
+import json from "~packages/util/json"
+import { requestWebAuthn } from "~packages/webAuthn/background/webAuthn"
+import type { BytesLike, HexString, UrlB64String } from "~typing"
+
+import type { WebAuthnValidatorOwner } from "../validator"
+
+export class WebAuthnOwner implements WebAuthnValidatorOwner {
+  public credentialId: UrlB64String
+
+  public use(credentialId: UrlB64String) {
+    this.credentialId = credentialId
+  }
+
+  public async sign(
+    challenge: BytesLike,
+    metadata?: {
+      sender?: browser.Runtime.MessageSender
+    }
+  ) {
+    if (!this.credentialId) {
+      throw new Error("Credential id is not set")
+    }
+
+    const challengeUrlB64 = isoBase64URL.fromBuffer(
+      typeof challenge === "string"
+        ? challenge.startsWith("0x")
+          ? isoUint8Array.fromHex(challenge.slice(2))
+          : isoUint8Array.fromHex(challenge)
+        : challenge
+    )
+
+    const { result: webAuthnAuthenticationPromise, cancel } =
+      await requestWebAuthn({
+        credentialId: this.credentialId,
+        challenge: challengeUrlB64
+      })
+
+    if (metadata?.sender) {
+      const senderOnRemovedHandler = async (tabId: number) => {
+        if (tabId !== metadata.sender.tab.id) {
+          return
+        }
+        await cancel()
+        browser.tabs.onRemoved.removeListener(senderOnRemovedHandler)
+      }
+      browser.tabs.onRemoved.addListener(senderOnRemovedHandler)
+    }
+
+    const webAuthnAuthentication = await webAuthnAuthenticationPromise
+    console.log(
+      `[passkeyOwnerWebAuthn] webAuthnAuthentication: ${json.stringify(
+        webAuthnAuthentication,
+        null,
+        2
+      )}`
+    )
+
+    const authData = parseAuthenticatorData(
+      Uint8Array.from(
+        Buffer.from(webAuthnAuthentication.authenticatorData.slice(2), "hex")
+      )
+    )
+
+    const authnticatorDataJson = {
+      rpIdHash: Buffer.from(authData.rpIdHash).toString("hex"),
+      flags: authData.flags,
+      counter: authData.counter
+    }
+
+    const signature = ethers.AbiCoder.defaultAbiCoder().encode(
+      ["uint256", "uint256"],
+      [webAuthnAuthentication.signature.r, webAuthnAuthentication.signature.s]
+    )
```

Comment:
Q: Shall we keep separated `r` and `s` and let the account decide their encoding way?

Reply:
It's sounds rational.

From the point of Passkey Account, the valid format of signature is (whether in P256Owner or WebAuthnOwner):
```solidity
    const signature = ethers.AbiCoder.defaultAbiCoder().encode(
      [
        "bool",
        "bytes",
        "bool",
        "string",
        "uint256",
        "uint256",
        "uint256",
        "uint256"
      ],
      [
        false,
        Uint8Array.from(Buffer.from(authenticatorData, "hex")),
        true,
        clientDataJson,
        clientDataJsonChallengeIndex,
        clientDataJsonTypeIndex,
        r,
        s
      ]
    )
```

But in imAccount, the valid format of signature is (whether in which validator):
```
ethers.AbiCoder.defaultAbiCoder().encode(
      [
        "address",
        "bytes",
        "(bytes authenticatorFlagsAndSignCount,string postChallengeData)"
      ],
      [validatorAddress, signature, webAuthnInput]
    )
```

The position of **WebAuthnValidator** is like the **PasskeyAccount**:

- The **WebAuthnValidator** and **PasskeyAccount** should construct the valid format of signature. 
- The responsibility of **Owner**(whether in WebAuthnOwner or P256Owner) is return the valid signature, which will be `r` and `s`.

So, the **WebAuthnValidator** and **PasskeyAccount** get the valid params (`r`, `s`, and others requirements) from **Owner**, and construct the valid signature for UserOp.

In short, I think the **Owner** returns `r` and `s` will be more flexible for **WebAuthnValidator** and **PasskeyAccount**.